cmd/buildkitd/main_oci_worker.go

```diff
@@ -89,32 +89,30 @@ func ociWorkerInitializer(c *cli.Context, common workerInitializerOpt) ([]worker
 	return []worker.Worker{w}, nil
 }
 
-func snapshotterFactory(name string) (runc.SnapshotterFactory, error) {
+func snapshotterFactory(commonRoot, name string) (runc.SnapshotterFactory, error) {
+	if name == "auto" {
```

Comment:
follow-up: the `auto` setting should probably also check if there is already some state for an existing snapshotter and choose the one that has state.